R/deprecated.R

```diff
@@ -114,12 +114,16 @@ deprecate_soft <- function(when,
 }
 
 #' @rdname deprecate_soft
+#' @param always If `FALSE`, the default, will warn every 8 hours.
+#'   If `TRUE`, will always warn. Only use `always = TRUE` after at least
+#'   one release with the default.
 #' @export
 deprecate_warn <- function(when,
                            what,
                            with = NULL,
                            details = NULL,
                            id = NULL,
+                           always = FALSE,
```

Comment:
Should we make this `frequency = c("always", "regularly", "once")` as in `rlang::warn()`?

Reply:
I don't _think_ so — that feels too much like tying this interface to its implementation. (Not that lifecycle currently uses `frequency` but we could now refactor to do so).

I think the goal here is to create a hierarchy of annoyingness that we can gradually move functions along over time. I could imagine something more like `annoy = c("a little", "a lot")` but we'd need to think about that more, and adding one extra step seems like enough for now.

@DavisVaughan any thoughts?